dex/dcr/script.go

```diff
@@ -539,6 +539,31 @@ func ExtractScriptHashByType(scriptType DCRScriptType, pkScript []byte) ([]byte,
 	return redeemScript, nil
 }
 
+// ExtractScriptData extracts script type, addresses, and required signature
+// count from a pkScript. Non-standard scripts are not necessarily an error;
+// non-nil errors are only returned if the script cannot be parsed. See also
+// InputInfo for additional signature script size data
+func ExtractScriptData(script []byte, chainParams *chaincfg.Params) (DCRScriptType, []string, int, error) {
+	class, addrs, numRequired, err := txscript.ExtractPkScriptAddrs(0, script, chainParams)
+	if err != nil {
+		return ScriptUnsupported, nil, 0, err
+	}
+	if class == txscript.NonStandardTy {
+		return ScriptUnsupported, nil, 0, nil
+	}
+
+	// Could switch on class from ExtractPkScriptAddrs, but use ParseScriptType
+	// for internal consistency.
+	scriptType := ParseScriptType(CurrentScriptVersion, script, nil)
```

Comment:
Similar to input info, but the added signature size computation and the binning of addresses into p2pkh and non-p2pkh wasn't needed.  Also, I wanted to do this without `ParseScriptType` since the `class` from `ExtractPkScriptAddrs` tells almost everything we need to know.  In fact I did this initially, but I felt it was best for now to stay consistent with the ParseScriptType behavior.